### ISSUE TITLE
Basic overview for copying

### ISSUE THREAD

**Issue by araspik** [jemalloc/jemalloc#1290]
I'm trying to implement at least a small subset of jemalloc (for a small standard library I'm making in [D](https://dlang.org) and I am unable to find even a small but recent overview of how jemalloc works. I've looked at `arena_s`, but I'm not quite sure what it's doing. I did read somewhere about it being similar to a pool, but I can't understand it. I've looked at `bin_s`, and I understand that slab allocation is taking place, but I can't understand the dynamics of the situation. Is there any explanation of how jemalloc works overall?

**Comment by davidtgoldblatt** [comment]
I don't think that there is a good overview of the recent design, unfortunately. The most up-to-date information is a facebook blog post here: https://www.facebook.com/notes/facebook-engineering/scalable-memory-allocation-using-jemalloc/480222803919/ .
